File: app/src/main/java/cn/nekocode/baseframework/utils/K.java
package cn.nekocode.baseframework.utils;

import android.content.Context;

import com.esotericsoftware.kryo.Kryo;
import com.esotericsoftware.kryo.io.Input;
import com.esotericsoftware.kryo.io.Output;

import java.io.FileInputStream;
import java.io.FileOutputStream;

/**
 * Created by Nekocode on 2014/11/24 0024.
 */
public class K {
    private static Kryo kryo = new Kryo();

    //文件名储存在这里
    public static String K_USER_BEAN = "user_bean";

    public static void save(Context context, String fileName, Object obj) {
        try {
            FileOutputStream fo = context.openFileOutput(fileName, Context.MODE_PRIVATE);
            Output output = new Output(fo);

            kryo.writeObject(output, obj);
            output.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public static <E>E load(Context context, String fileName, Class<E> c) {
        E rlt = null;
        try {
            FileInputStream fi = context.openFileInput(fileName);
            Input input = new Input(fi);
            rlt = kryo.readObject(input, c);
            input.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
        return rlt;
    }

    public static void delete(Context context, String fileName) {
        context.deleteFile(fileName);
    }
}

File: app/src/main/java/cn/nekocode/baseframework/utils/K.kt
package cn.nekocode.baseframework.utils

import android.content.Context
import cn.nekocode.baseframework.App
import cn.nekocode.baseframework.model.Weather
import cn.nekocode.baseframework.rest.REST
import com.esotericsoftware.kryo.Kryo
import com.esotericsoftware.kryo.io.Input
import com.esotericsoftware.kryo.io.Output
import rx.lang.kotlin.toSingletonObservable
import java.io.File
import kotlin.properties.Delegates

/**
 * Created by nekocode on 2015/8/13.
 */
public class K {
    companion object {
        private val kryo: Kryo = Kryo()
        val map: MutableMap<Any, Any> by Delegates.lazy {
            hashMapOf<Any, Any>()
        }

        fun write(fileName: String, obj: Any) {
            try {
                val fo = App.instance.openFileOutput(fileName, Context.MODE_PRIVATE)
                val output = Output(fo)

                kryo.writeObject(output, obj)
                output.close()
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }

        fun <E> read(fileName: String, c: Class<E>): E? {
            var rlt: E? = null
            try {
                val fi = App.instance.openFileInput(fileName)
                val input = Input(fi)
                rlt = kryo.readObject(input, c)
                input.close()
            } catch (e: Exception) {
                e.printStackTrace()
            }

            return rlt
        }

        fun rm(fileName: String) {
            App.instance.deleteFile(fileName)
        }

        fun rmAll() {
            val toDelete = File("/data/data/" + App.instance.getPackageName() + "/files")

            if (toDelete != null && toDelete.exists() && toDelete.isDirectory()) {
                for (item in toDelete.listFiles()!!) {
                    item.delete()
                }
            }
        }
    }
}
